miio/airhumidifier_miot.py

```diff
@@ -0,0 +1,357 @@
+import enum
+import logging
+from typing import Any, Dict, Optional
+
+import click
+
+from .click_common import EnumType, command, format_output
+from .exceptions import DeviceException
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+_MAPPING = {
+    # Source http://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:humidifier:0000A00E:zhimi-ca4:2
+    # Air Humidifier (siid=2)
+    "power": {"siid": 2, "piid": 1},  # bool
+    "fault": {"siid": 2, "piid": 2},  # [0, 15] step 1
+    "mode": {"siid": 2, "piid": 5},  # 0 - Auto, 1 - lvl1, 2 - lvl2, 3 - lvl3
+    "target_humidity": {"siid": 2, "piid": 6},  # [30, 80] step 1
+    "water_level": {"siid": 2, "piid": 7},  # [0, 128] step 1
+    "dry": {"siid": 2, "piid": 8},  # bool
+    "use_time": {"siid": 2, "piid": 9},  # [0, 2147483600], step 1
+    "button_pressed": {"siid": 2, "piid": 10},  # 0 - none, 1 - led, 2 - power
+    "speed_level": {"siid": 2, "piid": 11},  # [200, 2000], step 10
+    # Environment (siid=3)
+    "temperature": {"siid": 3, "piid": 7},  # [-40, 125] step 0.1
+    "fahrenheit": {"siid": 3, "piid": 8},  # [-40, 257] step 0.1
+    "humidity": {"siid": 3, "piid": 9},  # [0, 100] step 1
+    # Alarm (siid=4)
+    "buzzer": {"siid": 4, "piid": 1},
+    # Indicator Light (siid=5)
+    "led_brightness": {"siid": 5, "piid": 2},  # 0 - Off, 1 - Dim, 2 - Brightest
+    # Physical Control Locked (siid=6)
+    "child_lock": {"siid": 6, "piid": 1},  # bool
+    # Other (siid=7)
+    "actual_speed": {"siid": 7, "piid": 1},  # [0, 2000] step 1
+    "power_time": {"siid": 7, "piid": 3},  # [0, 4294967295] step 1
+}
+
+
+class AirHumidifierMiotException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    Auto = 0
+    Low = 1
+    Mid = 2
+    High = 3
+
+
+class LedBrightness(enum.Enum):
+    Off = 0
+    Dim = 1
+    Bright = 2
+
+
+class PressedButton(enum.Enum):
+    No = 0
+    Led = 1
+    Power = 2
+
+
+class AirHumidifierMiotStatus:
+    """Container for status reports from the air humidifier."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    # Air Humidifier
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if device is on."""
+        return self.data["power"]
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return "on" if self.is_on else "off"
+
+    @property
+    def fault(self) -> int:
+        """Fault state."""
+        return self.data["fault"]
```

Comment:
Should this be named `error`? Are there any known fault values?

Reply:
Nope, in API just wrote about 0 to 15 is possible value. 
Error maybe better yes, agree